src/azul/chalice.py

```diff
@@ -145,3 +159,122 @@ def _log_response(self, response):
     # Some type annotations to help with auto-complete
     lambda_context: LambdaContext
     current_request: Request
+
+
+class ValidatingAzulChaliceApp(AzulChaliceApp):
+
+    def create_spec_validators(self, resource_path, http_method) -> Mapping[str, SpecValidation]:
+        spec_validators: Dict[str, SpecValidation] = dict()
+        for parameter in self._extract_parameters(resource_path, http_method):
+            temp = SpecValidation(**parameter)
+            if 'schema' in parameter:
+                parameter_schema = parameter['schema']
+            elif 'content' in parameter:
+                for k, v in parameter['content'].items():
+                    parameter_schema = v['schema']
+            temp.validator = schema.type_lookup(parameter_schema['type'])
+            spec_validators[temp.name] = temp
+        return spec_validators
+
+    def validate_params(self) -> None:
+        def fmt_error(err_description, parameters):
+            joined = ', '.join(f'{p!r}' for p in parameters)
+            return f'{err_description} {pluralize("parameter", len(parameters))} {joined}'
+
+        # OpenAPI requires HTTP method names be lower case
+        http_method = self.current_request.method.lower()
+        resource_path = self.current_request.context['resourcePath']
+
+        query_params = self.current_request.query_params if self.current_request.query_params else {}
+        uri_params = self.current_request.uri_params if self.current_request.uri_params else {}
+        all_provided_params = {**query_params, **uri_params}
+
+        provided_params = set(all_provided_params.keys())
```

Comment:
It looks like you never addressed this, but still gave the thumbs up :thinking: 

> The set() call is unnecessary here, since a dictionary view object implements most (all?) set methods. Also I would probably just inline this variable.

https://github.com/DataBiosphere/azul/pull/1849/files#r471770897